database/engine/rrdengineapi.c

```diff
@@ -171,6 +176,10 @@ void rrdeng_store_metric_flush_current_page(RRDDIM *rd)
 
     handle = &rd->state->handle.rrdeng;
     ctx = handle->ctx;
+    if (ctx == NULL && rd->rrdset->rrdhost != localhost)
+            ctx = localhost->rrdeng_ctx;
```

Comment:
Since this check has  been done in `rrdeng_store_metric_init()` and the correct value has been assigned in `handle->ctx` can't we remove those 2 lines from `rrdeng_store_metric_flush_current_page()`, `rrdeng_store_metric_next()` and `rrdeng_store_metric_finalize()`?

Reply:
I will check it !